src/codeu/chat/client/core/View.java

```diff
@@ -136,4 +139,26 @@ public View(ConnectionSource source) {
 
     return messages;
   }
+  /**
+   * Get the info of the server; version info should be returned currently.
+   * @return The infomation of the server. If fails, null will be returned.
+   */
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
+      } else {
+        // Communicate this error - the server did not respond with the type of
+        // response we expected.
+        new IOException("ERROR: The server did not respond with the type of response we expected.").printStackTrace();
+      }
+    } catch (Exception ex) {
+      // Communicate this error - something went wrong with the connection.
+      new UnexpectedException("FATAL ERROR: Unexpected error occurs, something went wrong while client was trying to connect to server.").printStackTrace();
```

Comment:
I understand what sort of format you are trying to do here, but here's the format in which it should be done:
catch (Exception ex) {
System.out.println("<insert error message here>");
ex.getStackTrace();

Also, it's important to note that this is not a fatal or unexpected error, or else the comments would have told us so.